File: 0852-peak-index-in-a-mountain-array/0852-peak-index-in-a-mountain-array.java
class Solution {
    public int peakIndexInMountainArray(int[] arr) {
        int mid=0;
        int low=0;
        int high=arr.length-1;
        while(low<high){
            mid=low+(high-low)/2;
            if(arr[mid]<arr[mid+1]){
                low=mid+1;
            }
            else {
                high=mid;
            }
        }
        return high;
    }
}

File: 0852-peak-index-in-a-mountain-array/0852-peak-index-in-a-mountain-array.py
class Solution:
    def peakIndexInMountainArray(self, nums: List[int]) -> int:
        i=0
        j=len(nums)-1
        while(i<j):
            mid=i+(j-i)//2
            if nums[mid]>nums[mid+1]:
                j=mid
            else:
                i=mid+1
        return i
